### ISSUE TITLE
Doesn't save error messages properly when built with `ARGS_NOEXCEPT`

### ISSUE THREAD

**Issue by alex65536** [Taywee/args#105]
Consider the following code:

```cpp
#define ARGS_NOEXCEPT
#include "args.hxx"

int main(int argc, char **argv) {
    args::ArgumentParser parser("This is a test program.", "Test description.");
    args::HelpFlag help(parser, "help", "Display this help menu", {'h', "help"});
    args::ValueFlag<int> first(parser, "VALUE", "First value", {'f', "first"}, args::Options::Required);
    args::ValueFlag<int> second(parser, "VALUE", "Second value", {'s', "second"}, 42);
    
    parser.ParseCLI(argc, argv);
    if (parser.GetError() == args::Error::Help) {
        std::cout << parser << std::endl;
        return 0;
    }
    if (parser.GetError() != args::Error::None) {
        std::cout << "Error: " << (int)parser.GetError() << " " << parser.GetErrorMsg() << std::endl;
        return 1;
    }

    std::cout << "Got: " << first.Get() << " " << second.Get() << std::endl;
    return 0;
}
```

When I run the program without specifying `-f` option, it just print `Error: `, but the contents of `GetErrorMsg()` are empty. When I rewrite the code with exceptions, it works fine.

**Comment by Taywee** [comment]
Thanks for the report (and especially for the example code).  The library was written first and foremost with exceptions in mind, and the noexception operation was thrown in after the fact (and quite a few versions ago).  I wish I had done it the other way around.  I'm pretty unhappy with the state of exceptionless operation in args (having to specify a preprocessor define is very ugly).

I'll try to fix this issue soon.  I'd like to fully revise the noexception API in a future major release and make it significantly more sane.

**Comment by Trider12** [comment]
Having the same issue.

**Comment by Taywee** [comment]
And the "reader" concept makes it nearly impossible to actually report strings for ParseErrors, because that template parameter needs to support just being a function pointer.  I'll try to see if I can get more string errors wired through, but it won't be able to be complete without breaking API changes.

**Comment by Taywee** [comment]
This particular case is fixed as of 6.4.2.  I can't guarantee 100% message coverage with `ARGS_NOEXCEPT` use, but this should cover these common cases.
